lmod/SitePackage_warnings.lua

```diff
@@ -0,0 +1,141 @@
+To test your jobs with the new environment, please run:
+module load StdEnv/2020
+
+To change your default version immediately, please run the following command:
+
+echo "module-version StdEnv/2020 default" >> $HOME/.modulerc
+
+For more information, please see:
+https://docs.computecanada.ca/wiki/Standard_software_environments]])
+		end
+		--color_banner("red")
+	end
+	
+	-- only show the warning if the user provided a shortened version of python as load, if the defaultKind is system or marked, and if it does not result in 3.10.2
+	if (userProvidedName == "python" or userProvidedName == "python/3" or userProvidedName == "python/3.") then
+		if (moduleVersion ~= "3.10.2" and (defaultKind == "system" or defaultKind == "unknown" or defaultKind == "marked")) then
+			if (string.sub(lang,1,2) == "fr") then
+				LmodWarning([[Attention, le 4 avril 2023, la version par défaut de python deviendra 3.10. 
+Pour continuer d'utiliser la version 3.8, veuillez charger le module python/3.8 explicitement.
+]])
+			else
+				LmodWarning([[Warning. On April 4th 2023, the default version of python will become 3.10. 
+To keep using python 3.8, please load the python/3.8 module explicitly. 
+]])
+			end
+		end
+		--color_banner("red")
+	end
+end
+
+function incomplete_version_warning(t)
+   local min_digits = {
+      [ { "python", "arrow" } ]       = 2,
+      [ { "scipy-stack", "StdEnv" } ] = 1,
+   }
+
+   for modules,min_digit in pairs(min_digits) do
+     ------------------------------------------------------------
+     -- Look for fullName first otherwise sn
+     if (has_value(modules,myModuleFullName()) or has_value(modules,myModuleName())) then
+        ----------------------------------------------------------
+	local FrameStk   = require("FrameStk")
+	local frameStk   = FrameStk:singleton()
+	local userProvidedName = frameStk:userName()
+	local userProvidedVersion = string.match(userProvidedName, "/(.*)") or ""
+	local num_digit = 0
+	if userProvidedVersion == "" then
+		num_digit = 0
+	else
+		local _, num_dot = string.gsub(userProvidedVersion, "%.", "")
+		num_digit = num_dot+1
+	end
+	if (num_digit < min_digit) then
+   		local lang = os.getenv("LANG") or "en"
+		if (string.sub(lang,1,2) == "fr") then
+			LmodWarning([[Attention, vous avez chargé le module ]] .. myModuleName() .. [[ en spécifiant une version incomplète: "]] .. userProvidedVersion .. [[".
+Nous vous recommandons fortement de spécifier au moins ]] .. min_digit .. [[ chiffres pour la version de ce module. Dans le cas contraire, une future mise à jour
+de ce module pourrait faire échouer vos tâches.]])
+		else
+			LmodWarning([[Warning, you have loaded the module ]] .. myModuleName() .. [[ by specifying an incomplete version: "]] .. userProvidedVersion .. [[".
+We strongly recommend that you specify at least ]] .. min_digit .. [[ digits for the version of this module. Not doing this could crash your jobs when we install a newer version in the future.]])
```

Comment:
I'd add a word about reproducibility of jobs/submission scripts as it a strong argument along with job crashes.

Something like?
`... Not doing this could crash or make your jobs non-reproducible when we install a newer version in the future.